impl/impl.go

```diff
@@ -331,134 +321,122 @@ func (m *manager) updateValidationStatus(ctx context.Context, chid datatransfer.
 		return err
 	}
 
-	// dispatch channel events and generate a response message
-	chst, response, err := m.processValidationUpdate(ctx, chid, result)
-
-	// dispatch transport updates
-	return m.transport.UpdateChannel(ctx, chid, datatransfer.ChannelUpdate{
-		Paused:      result.LeaveRequestPaused(chst),
-		Closed:      err != nil || !result.Accepted,
-		SendMessage: response,
-	})
-}
-
-func (m *manager) processValidationUpdate(ctx context.Context, chid datatransfer.ChannelID, result datatransfer.ValidationResult) (datatransfer.ChannelState, datatransfer.Response, error) {
-
 	// read the channel state
 	chst, err := m.channels.GetByID(context.TODO(), chid)
 	if err != nil {
-		return nil, nil, err
+		return err
 	}
 
-	// if the request is now rejected, error the channel
-	if !result.Accepted {
-		err = m.recordRejectedValidationEvents(chid, result)
-	} else {
-		err = m.recordAcceptedValidationEvents(chst, result)
-	}
+	// dispatch channel events and generate a response message
+	err = m.processValidationUpdate(ctx, chst, result)
```

Comment:
Not sure this refactor buys you anything tbh, it's fine, but how much I miss ternaries in Go! In a sane programming language this would simply be:

```go
err = !result.Accepted ? m.recordRejectedValidationEvents(chid, result) : m.recordAcceptedValidationEvents(chst, result)
```

Reply:
I hate the lack of terneries in go too. on this we agree!